eclipse-theia/theia
# yarn upgrade

#### What it does

Upgrade all our dependencies, according to our already specified ranges.

I had to tighten some constraints because the latest version of some
packages cause issues. Notably, the typescript compiler has an issue on
3.6.3 but not 3.5.3, so I constrained it such as it wouldn't ever match
the version that causes issues.

In the long run, we should be more careful with the version ranges we
use, especially with runtime dependencies.

Fixes https://github.com/eclipse-theia/theia/issues/6500
~~DoNotFixes https://github.com/eclipse-theia/theia/issues/6529~~

#### How to test

- Build should pass.
- Tests should pass.
- Try anything and everything in the example applications to see if anything is broken.

Keep in mind that this commit only updates the lockfile, which means that we will use packages closer to what a fresh Theia build would pull, instead of the outdated versions pinned in the lockfile for the main repository.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
